packages/admin/cms-admin/src/pages/pageTree/PageTree.tsx

```diff
@@ -60,6 +63,33 @@ const PageTree: React.ForwardRefRenderFunction<PageTreeRefApi, PageTreeProps> =
     { pages, editDialogApi, toggleExpand, onSelectChanged, category, siteUrl },
     ref,
 ) => {
+    const client = useApolloClient();
+    const newPageIds = React.useRef<string[]>([]);
+
+    const queries = client.getObservableQueries();
+    const pagesQuery = Array.from(queries.values()).find((query) => query.queryName === namedOperations.Query.Pages) as
+        | ObservableQuery<GQLPagesQuery, GQLPagesQueryVariables>
+        | undefined;
+
+    if (pagesQuery) {
+        client.cache.watch<GQLPagesQuery, GQLPagesQueryVariables>({
```

Comment:
Is it safe to call this function on every render? I'd expect some sort of unsubscribe mechanism here

Reply:
You are right, that's not optimal. I moved it to a `useEffect()` hook